base/test-interfaces/src/test/java/org/openscience/cdk/interfaces/AbstractStrandTest.java

```diff
@@ -236,7 +232,7 @@ public void testToString() {
     public void testClone() throws Exception {
         IStrand strand = (IStrand) newChemObject();
         Object clone = strand.clone();
-        Assert.assertTrue(clone instanceof IStrand);
+        Assert.assertNotNull(clone);
```

Comment:
This one I do not get. Is it tested at compile and/or runtime at a Java level that the clone must be of the same interface? The `testClone()` is making sure the cloned class has the right interface. Does Java guarantee that, that this test is not needed?

Reply:
> is IStrand.clone() required to always return something of IStrand, at the expense of a Java RuntimeError or compile error?

Yes it's static typing (i.e. a compile error).

``` java
public class SuperClass extends BaseClass {
    B method() {
        return new B();
    }
}

class BaseClass {
    A method() {
        return new A();
    }
}

class A { }

class B { }

class C extends A { }

```

```
harbinger:tmp john$ javac SuperClass.java 
SuperClass.java:2: error: method() in SuperClass cannot override method() in BaseClass
    B method() {
      ^
  return type B is not compatible with A
1 error
```

Reply:
OK, but doesn't the `clone()` API always return an `Object`??

Reply:
The base does... but since all Objects extend 'Object' you can change the return type as has been done in the CDK. In fact you're also meant to get rid of the CloneNotSupportedException (if it can't occur) but unfortunately they wasn't done in the CDK so we're stuck with it now.